protocol/messenger_backup_handler.go

```diff
@@ -304,22 +304,27 @@ func (m *Messenger) handleWatchOnlyAccount(message *protobuf.SyncAccount) error
 
 func (m *Messenger) handleSyncedCommunities(state *ReceivedMessageState, message *protobuf.Backup) []error {
 	var errors []error
+	handledCommunities := make(map[string]struct{})
 	for _, syncCommunity := range message.Communities {
+		idString := string(syncCommunity.GetId())
+
 		err := m.handleSyncInstallationCommunity(state, syncCommunity, nil)
 		if err != nil {
 			errors = append(errors, err)
 		}
 
-		err = m.requestCommunityKeys(state, syncCommunity)
+		_, alreadyHandled := handledCommunities[idString]
+		err = m.requestCommunityKeysAndSharedAddresses(state, syncCommunity, alreadyHandled)
```

Comment:
Do we want to request shared addresses each time we handle backup, even outdated one?

Reply:
Don't we handle backup only once when we restore the account?

Reply:
Right, if that's really the case, I'll indeed need to update the code, though we'll need to update the part below too as we request keys for communities too

Reply:
Oh, true, I tough we are checking if we have missing keys before we ask but seems we don't :thinking:

IMO, asking for keys and shared addresses on backup is fine, even if it's redundant, but we should do it at most once. It is fine for me to address in separate issue :+1:

Reply:
I checked the code and the `ProcessBackedupMessages` property is set yo true in this condition: `mnemonic != "" && fetchBackup`, so it seems to only happen on re-import. @saledjenic or @igor-sirotin can confirm

Reply:
I know :sweat_smile:

> I assume, this is only the case when we are recovering account from seedphrase, but even though, it means backup messages processing is enabled for that login session.

Reply:
> it means backup messages processing is enabled for that login session.

Ah right, I missed that part. Indeed, in that case, we probably don't want to send further requests. I'm not sure who to addresses that though. I can open an issue and we can think about it

Reply:
Issue opened here: https://github.com/status-im/status-go/issues/5361